cli/azd/pkg/apphost/generate.go

```diff
@@ -423,6 +502,104 @@ func (b *infraGenerator) addServiceBus(name string, queues, topics *[]string) {
+	pValue := comp.Value
+
+	if matched, _ := regexp.MatchString(`{[a-zA-Z][a-zA-Z0-9]*\.inputs\.[a-zA-Z][a-zA-Z0-9]*}`, pValue); !matched {
+		// parameters not using inputs are not bicep input parameters
+		return fmt.Errorf("parameter %s does not use inputs. This is currently not supported.", name)
+	}
+
+	valueParts := strings.Split(
+		strings.TrimRight(strings.TrimLeft(pValue, "{"), "}"),
+		".inputs.")
+	// regex from above ensure parts 0 and 1 exists
+	resourceName, inputName := valueParts[0], valueParts[1]
+
+	if name != resourceName {
+		return fmt.Errorf("parameter %s does not use inputs from its own resource. This is not supported", name)
+	}
+
+	input, exists := comp.Inputs[inputName]
+	if !exists {
+		return fmt.Errorf("parameter %s does not have input %s", name, inputName)
+	}
+
+	b.bicepContext.InputParameters[name] = input
+	return nil
+}
+
+func (b *infraGenerator) addBicep(name string, comp *Resource) error {
+	if comp.Path == nil {
+		if comp.Parent == nil {
+			return fmt.Errorf("bicep resource %s does not have a path or a parent", name)
+		}
+		// module uses parent
+		return nil
+	}
+	if comp.Params == nil {
+		comp.Params = make(map[string]any)
+	}
+
+	// afterInjectionParams is used to know which params where actually injected
+	autoInjectedParams := make(map[string]any)
+	// params from resource are type-less (any), injectValueForBicepParameter() will convert them to string
+	// by converting to string, we can evaluate arrays and objects with placeholders.
+	stringParams := make(map[string]string)
+	for p, pVal := range comp.Params {
+		paramValue, injected, err := injectValueForBicepParameter(name, p, pVal)
+		if err != nil {
+			return fmt.Errorf("injecting value for bicep parameter %s: %w", p, err)
+		}
+		stringParams[p] = paramValue
+		if injected {
+			autoInjectedParams[p] = struct{}{}
+		}
+	}
+	if _, keyVaultInjected := autoInjectedParams["keyVaultName"]; keyVaultInjected {
+		b.addKeyVault(name+"kv", true, true)
+	}
+
+	b.bicepContext.BicepModules[name] = genBicepModules{Path: *comp.Path, Params: stringParams}
+	return nil
+}
+
+// injectValueForBicepParameter checks for aspire-manifest and azd conventions rules for auto injecting values for
+// the bicep.v0 parameters.
+// Conventions examples:
+// - for a `keyVaultName` parameter, set the value to the output of the keyVault resource to be created.
+// - for `principalName`, set the value to the managed identity created by azd.
+// Note: The value is only injected when it is an empty string.
+// injectValueForBicepParameter returns the final value for the parameter, a boolean indicating if the value was injected
+// and an error if any.
+func injectValueForBicepParameter(resourceName, p string, parameter any) (string, bool, error) {
+	// using json.Marshal to parse any type of value (array, bool, etc)
+	jsonBytes, err := json.Marshal(parameter)
+	finalParamValue := string(jsonBytes)
+	if err != nil {
```

Comment:
I think this is safe -- no nil panic, but consider using `jsonBytes` after the error checking...